### PR TITLE
1.8.7 compatibility fix

### PR DESCRIPTION
**Pull request by loadedsith**
This works for me in 1.8.7 and my test ruby 2.0.0p247.

### CODE REVIEW

File: lib/succubus/generator.rb [hunk 1]
```diff
@@ -69,14 +70,16 @@ def invoke(rule)
       # Scan it for all instances of (unescaped) <rules>
       # Each match will be an array containing the portion of string before
       # the <rule>; and the <rule> itself (which may be empty on the last match)
-      choice.scan(/(.*?(?<!\\)(?:\\\\)*)(<.*?>|$)/) do |match|
+
+      choice.scan(/(.*?)(<.*?>|$)/) do |match|
```
Comment (asilano) [inline lib/succubus/generator.rb]: This is a functional change. While I haven't documented it, the existing code prevents the generator expanding backslash-escaped groups such as `"Don't expand this \<group>"`. I see Regex lookbehinds don't work in 1.8.7, so we'd need another tactic here.

Comment (loadedsith) [inline lib/succubus/generator.rb]: I'll see if I can get that behavior tonight, unless you beat me to it

On 7/19/13, Chris Howlett notifications@github.com wrote:

> > @@ -69,14 +70,16 @@ def invoke(rule)
> >        # Scan it for all instances of (unescaped) <rules>
> >        # Each match will be an array containing the portion of string
> > before
> >        # the <rule>; and the <rule> itself (which may be empty on the last
> > match)
> > -      choice.scan(/(._?(?<!\)(?:\\)_)(<.*?>|$)/) do |match|
> >   +
> > -      choice.scan(/(._?)(<._?>|$)/) do |match|
> 
> This is a functional change. While I haven't documented it, the existing
> code prevents the generator expanding backslash-escaped groups such as
> `"Don't expand this \<group>"`. I see Regex lookbehinds don't work in 1.8.7,
> so we'd need another tactic here.
> 
> ---
> 
> Reply to this email directly or view it on GitHub:
> https://github.com/asilano/succubus/pull/3/files#r5294831

## 

Graham P Heath

**Comment by loadedsith** [comment]
How does /(._?)([^\]<._?>|$)/ work for you?
screen shot 2013-07-19 at 12 09 37 pm
screen shot 2013-07-19 at 12 09 43 pm

**Comment by loadedsith** [comment]
Never mind, its not matching spaces, will fix and repost

**Comment by asilano** [comment]
If I _ever_ get back to this, I won't be interested in targeting Ruby 1.8.x!

**Pull request closed without merging**